src/jit/decomposelongs.cpp

```diff
@@ -1030,6 +1030,51 @@ GenTree* DecomposeLongs::DecomposeMul(LIR::Use& use)
 }
 
 //------------------------------------------------------------------------
+// DecomposeUMod: Decompose GT_UMOD. The only GT_UMODs that make it to decompose
+// are guaranteed to be an unsigned long mod with op2 which is a cast to long from
+// a constant int whose value is between 2 and 0x3fffffff. All other GT_UMODs are
+// morphed into helper calls. These GT_UMODs will actually return an int value in
+// RDX. In decompose, we change the type of the operation to a TYP_INT, and replace
+// op2 with its low half.
+//
```

Comment:
It would be useful to also point out that you leave the op1 as a long, so you end up with:

GT_UMOD[TYP_INT] ( GT_LONG [TYP_LONG] (loOp1, hiOp1), loOp2 [TYP_INT] )

where we expect to generate:

EDX = hiOp1
EAX = loOp1
reg = loOp2
idiv reg
EDX is the remainder, and result of GT_UMOD